src/main/scala/org/allenai/deep_qa/data/BackgroundCorpusSearcher.scala

```diff
@@ -126,3 +146,33 @@ class LuceneBackgroundCorpusSearcher(params: JValue) extends BackgroundCorpusSea
     return true
   }
 }
+
+
+// trait used for handling various types of queries
+trait Query {}
+
+
+// Query type for basic string queries
+class StringQuery(var query: String = "") extends Query {
```

Comment:
There's no need for this class to be mutable.  Mutability should be avoided where it is not necessary.  So you can just do `case class StringQuery(query: String) extends Query`, with no additional class body necessary.